### PR TITLE
[wifi_info_flutter] [wifi_info_flutter_platform_interface] Initial commit for `wifi_info_flutter` plugin and platform interface

### PR DESCRIPTION
**Pull request by bparrishMines**
## Description

This PR is from the generated code when running the lines below. Only small changes were made to get tests passing. 

`flutter create --platforms=android,ios -t plugin --org="io.flutter.plugins" -a java -i objc --project-name=wifi_info_flutter .`
`flutter create -t package --project-name=wifi_info_flutter_platform_interface .`

Beginning a larger effort to separate wifi related methods from the `connectivity` plugin. See https://github.com/flutter/flutter/issues/40418

## Related Issues

https://github.com/flutter/flutter/issues/40418

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/wifi_info_flutter/wifi_info_flutter_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## [0.0.1] - TODO: Add release date.
```
Comment (amirh) [inline packages/wifi_info_flutter/wifi_info_flutter_platform_interface/CHANGELOG.md:1]: update

File: packages/wifi_info_flutter/wifi_info_flutter_platform_interface/LICENSE [hunk 2]
```diff
@@ -0,0 +1 @@
+TODO: Add your license here.
```
Comment (amirh) [inline packages/wifi_info_flutter/wifi_info_flutter_platform_interface/LICENSE:1]: update

File: packages/wifi_info_flutter/wifi_info_flutter_platform_interface/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# wifi_info_flutter_platform_interface
+
+A new Flutter package project.
```
Comment (amirh) [inline packages/wifi_info_flutter/wifi_info_flutter_platform_interface/README.md:3]: update

File: packages/wifi_info_flutter/wifi_info_flutter_platform_interface/lib/wifi_info_flutter_platform_interface.dart [hunk 4]
```diff
@@ -0,0 +1,7 @@
+library wifi_info_flutter_platform_interface;
+
+/// A Calculator.
+class Calculator {
```
Comment (amirh) [inline packages/wifi_info_flutter/wifi_info_flutter_platform_interface/lib/wifi_info_flutter_platform_interface.dart:4]: ?

File: packages/wifi_info_flutter/wifi_info_flutter_platform_interface/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+name: wifi_info_flutter_platform_interface
+description: A new Flutter package project.
+version: 0.0.1
```
Comment (amirh) [inline packages/wifi_info_flutter/wifi_info_flutter_platform_interface/pubspec.yaml:3]: Let's start the platform interface at 1 as we don't want major bumps

File: packages/wifi_info_flutter/wifi_info_flutter_platform_interface/pubspec.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  flutter_test:
+    sdk: flutter
+
+# For information on the generic Dart part of this file, see the
+# following page: https://dart.dev/tools/pub/pubspec
+
+# The following section is specific to Flutter.
+flutter:
+
+  # To add assets to your package, add an assets section, like this:
```
Comment (amirh) [inline packages/wifi_info_flutter/wifi_info_flutter_platform_interface/pubspec.yaml:24]: remove the boilerplate

File: packages/wifi_info_flutter/wifi_info_flutter_platform_interface/test/wifi_info_flutter_platform_interface_test.dart [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import 'package:flutter_test/flutter_test.dart';
```
Comment (amirh) [inline packages/wifi_info_flutter/wifi_info_flutter_platform_interface/test/wifi_info_flutter_platform_interface_test.dart:1]: ?

**Review by amirh — CHANGES_REQUESTED** [review]

**Comment by bparrishMines** [comment]
@amirh This PR was only supposed to be code generated from `flutter create` and small changes to get tests passing. Ideally, this makes it easier to review and the following PRs will have non-generated code.

**Comment by amirh** [comment]
Ok! let's just add the license files before submitting.

**Review by amirh — APPROVED** [review]
LGTM just add license files before submitting

**Pull request merged**